src/pages/plugins/plugins.js

```diff
@@ -285,61 +350,206 @@ export default function PluginsInclude(updates) {
+        return;
       }
-      
+
       const installed = await fsOperation(PLUGIN_DIR).lsDir();
       const disabledMap = settings.value.pluginsDisabled || {};
-      
+
       installed.forEach(({ url }) => {
-        const plugin = fetchedPlugins.find(({ id }) => id === Url.basename(url));
+        const plugin = items.find(({ id }) => id === Url.basename(url));
         if (plugin) {
           plugin.installed = true;
           plugin.enabled = disabledMap[plugin.id] !== true;
           plugin.onToggleEnabled = onToggleEnabled;
           plugin.localPlugin = getLocalRes(plugin.id, "plugin.json");
         }
       });
-      
-      // Add plugins to the all plugins array 
-      plugins.all.push(...fetchedPlugins);
-      
+
+      if (isInitial) {
+        $list.all.querySelectorAll(".filter-empty").forEach((el) => el.remove());
+      }
+
+      plugins.all.push(...items);
+
       const fragment = document.createDocumentFragment();
-      fetchedPlugins.forEach((plugin) => {
+      items.forEach((plugin) => {
         fragment.append(<Item {...plugin} updates={updates} />);
       });
-      $list.all.append(fragment);
-      
-      currentPage++;
-      $list.all.setAttribute("empty-msg", strings["no plugins found"]);
+
+      if (fragment.childNodes.length) {
+        $list.all.append(fragment);
+      } else if (isInitial) {
+        $list.all.append(
+          <div className="filter-empty">{strings["no plugins found"] || "No plugins found"}</div>,
+        );
+      }
+
+      hasMore = hasMoreResults;
+      if (!hasMore) {
+        $list.all.setAttribute("empty-msg", strings["no plugins found"]);
+      }
     } catch (error) {
       $list.all.setAttribute("empty-msg", strings["error"]);
-      window.log("error", "Failed to filter plugins:");
-      window.log("error", error);
+      console.error("Failed to filter plugins:", error);
+      hasMore = false;
     } finally {
       isLoading = false;
     }
   }
 
+  async function retrieveFilteredPlugins(filterState) {
+    if (!filterState) return { items: [], hasMore: false };
+
+    if (filterState.type === "orderBy") {
+      const page = filterState.nextPage || 1;
+      try {
+        let response;
+        if (filterState.value === "top_rated") {
+          response = await fetch(
+            `${constants.API_BASE}/plugins?explore=random&page=${page}&limit=${LIMIT}`,
+          );
+        } else {
+          response = await fetch(
+            `${constants.API_BASE}/plugin?orderBy=${filterState.value}&page=${page}&limit=${LIMIT}`,
+          );
+        }
+        const items = await response.json();
+        if (!Array.isArray(items)) {
+          return { items: [], hasMore: false };
+        }
+        filterState.nextPage = page + 1;
+        const hasMoreResults = items.length === LIMIT;
+        return { items, hasMore: hasMoreResults };
+      } catch (error) {
+        console.error("Failed to fetch ordered plugins:", error);
```

Comment:
`window.log` function usage is missing. That function outputs information to the `acode.log` file, Helpful for verbose/debugging purposes


```suggestion
        console.error("Failed to fetch ordered plugins:", error);
        window.log("error", "Failed to fetch ordered plugins:");
        window.log("error", error);
```